modules/mapbox/src/deck-utils.js

```diff
@@ -8,6 +8,11 @@ export function getDeckInstance({map, gl, deck}) {
 
   const customRender = deck && deck.props._customRender;
 
+  if (deck && deck.props.views && !deck.props.views.some(view => view.id === 'mapbox')) {
```

Comment:
You can't manipulate views during initialization. If the user defines views inline in React then it will be replaced.

Reply:
Ok, that makes sense. This can be moved into the `_drawLayers` calls